### PR TITLE
Fix supplying null entries into the creative list

### LINKED ISSUE #1171: Creative Mode crash with certain blocks disabled
Railcraft will crash using the Creative Search tab (not the Railcraft tab) with certain blocks disabled.  From the testing I did, these blocks were the ones at fault:

    B:ore_dark_diamond=false
    B:ore_dark_emerald=false
    B:ore_dark_lapis=false
    B:ore_saltpeter=false
    B:ore_sulfur=false

Other ores can still be disabled safely.

May be related to #1162.

edit: Ah, almost forgot a crash log: https://pastebin.com/su0dktz2

### PR DESCRIPTION
**Pull request by liach**
Fixed #1171 after the second commit. Before the game instantly crashes when you click that search tab. Now there is no effect, although loot table warns for missing items when the signal module is disabled (does not crash game)

### CODE REVIEW

File: src/main/java/mods/railcraft/common/util/misc/Predicates.java [hunk 1]
```diff
@@ -24,7 +25,13 @@
  *
  * @author CovertJaguar <http://www.railcraft.info>
  */
+@SuppressWarnings({"unchecked", "rawtypes"})
 public class Predicates {
+
+    private static final Predicate NON_NULL = Objects::nonNull;
+    private static final Predicate ALWAYS_FALSE = t -> false;
+    private static final Predicate ALWAYS_TRUE = t -> true;
```
Comment (CovertJaguar) [inline src/main/java/mods/railcraft/common/util/misc/Predicates.java:33]: You are breaking the generics, keep your PRs focused. Ignoring the warnings is stupid.

**Review by CovertJaguar — CHANGES_REQUESTED** [review]
Removes the changes to Predicates, and it looks good.

File: src/main/java/mods/railcraft/common/util/misc/Predicates.java (hunk 1 above)
Comment (liach) [reply src/main/java/mods/railcraft/common/util/misc/Predicates.java:33]: It does not break. You should use `rawtypes`, just like `java.util.Collections.emptyList()` returns a generic list casted from a rawtype list. [Link to the source](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Collections.java#l4458)

Comment (CovertJaguar) [reply src/main/java/mods/railcraft/common/util/misc/Predicates.java:33]: Why did you even make the change?

Comment (liach) [reply src/main/java/mods/railcraft/common/util/misc/Predicates.java:33]: To prevent creating extra objects.

**Review by CovertJaguar — APPROVED** [review]

**Pull request merged**